compiler/src/dotty/tools/dotc/transform/init/Checker.scala

```diff
@@ -36,12 +36,16 @@ class Checker extends Phase:
     traverser.traverse(unit.tpdTree)
 
   override def runOn(units: List[CompilationUnit])(using Context): List[CompilationUnit] =
-    val checkCtx = ctx.fresh.setPhase(this.start)
+    val checkCtx = ctx.fresh.setPhase(this)
```

Comment:
I am not sure this could have unintended consecuences.

Reply:
Previously we did something wrong --- otherwise, the tree and symbol info will not be in sync.